src/phonebook/views.py

```diff
@@ -59,16 +68,56 @@ def dump_phonebook(self, poc: bool) -> list[DectRegistration]:
             }
             if poc:
                 # POC member, include extra info
+                ipei = None
+                if dect.ipei:
+                    ipei = dectutil.format_ipei(dect.ipei[0], dect.ipei[1])
                 entry.update(
                     {
                         "activation_code": dect.activation_code,
                         "publish_in_phonebook": dect.publish_in_phonebook,
+                        "ipei": ipei,
                     },
                 )
             phonebook.append(entry)
         return phonebook
 
 
+@method_decorator(csrf_exempt, name="dispatch")
+class DectUpdateIPEI(
+    CampViewMixin,
+    ScopedProtectedResourceView,
+    View,
+):
+    """
+    Function to update IPEI after user registered their phone
+    """
+
+    required_scopes = ["phonebook:admin"]
+
+    def post(self, request, dect_number, *args, **kwargs):
+        team = Team.objects.get(name="POC", camp=self.camp)
+        if not self.request.user in team.leads and self.request.access_token.is_valid(
```

Comment:
this should check for the teamlead permission instead of checking for lead status of the actual team membership, maybe we have or can make a mixin for it